pkg/storage/stores/shipper/indexgateway/gateway.go

```diff
@@ -333,9 +333,14 @@ func (g *Gateway) LabelValuesForMetricName(ctx context.Context, req *indexgatewa
 	if err != nil {
 		return nil, err
 	}
-	matchers, err := syntax.ParseMatchers(req.Matchers)
-	if err != nil {
-		return nil, err
+	var matchers []*labels.Matcher
+	// An empty matchers string cannot be parsed,
+	// therefore we check the string representation of the the matchers.
+	if req.Matchers != syntax.EmptyMatchers {
+		matchers, err = syntax.ParseMatchers(req.Matchers)
+		if err != nil {
+			return nil, err
+		}
 	}
 	names, err := g.indexQuerier.LabelValuesForMetricName(ctx, instanceID, req.From, req.Through, req.MetricName, req.LabelName, matchers...)
```

Comment:
Hope it's ok to pass empty `[]*labels.Matcher` to `g.indexQuerier.LabelValuesForMetricName` when `req.Matchers == syntax.EmptyMatchers`

Also can you add a test?

Reply:
The index store checks for `len(matchers)`:
https://github.com/grafana/loki/blob/ea59e4bdc48ebc9c9ff17a9bf9dc01e6833f07fe/pkg/storage/stores/series/series_index_store.go#L239-L241